### DISCUSSION TITLE
Exclude retweets and quotes in parameters during search tweets

### DISCUSSION THREAD

**Discussion by 2jjk** [tweepy/tweepy#1801] [Questions] [answered]
Is there a way to exclude retweets and quotes from the search API? I'd like to have a checkbox where users can choose if they want to include retweets of a relevant keyword/topic. I don't want to filter them out after going through the tweets. I've seen the **-is:retweet** flag, but how would I use it as a parameter in my **client.search_recent_tweets** function? Or should it be concatenated to the query as "-is:retweet"? Not sure what the default behavior is or if it extends to quotes as well.

Another quick question; can I also include a location parameter? I want to only search (not filter after getting everything) for tweets originating from a certain country of users. I know this can depend on whether users are sharing their location, but is there any support for that? I can get all the tweets and the location from user.fields then filter but is there anything more convenient? 

Very new to this so any help is appreciated. Thanks.

**Comment by Harmon758** [comment, marked as answer]
You can add `-is:retweet` directly to your query (separated by a space).
Quote Tweets are different from Retweets, but you can also add `-is:quote` to your query.
There are several location operators, namely `place:`, `place_country:`, `point_radius:`, and `bounding_box:`.
See [the Twitter API documentation on all of this](https://developer.twitter.com/en/docs/twitter-api/tweets/search/integrate/build-a-query).

**Reply by 2jjk** [reply]
Thank you for the reply. I currently have the free project so I can't use parameters like the country. I can't apply for Academic Research access as I'm not a student nor a researcher. Can I use those parameters in other ways? As I only see the Advanced (Academic Research) option for them.

**Reply by Harmon758** [reply]
You're right; unfortunately, they're restricted to Academic Research access.
This isn't up to Tweepy. You should provide feedback through the [Twitter Developers Forums](https://twittercommunity.com/) or [Twitter Developer Feedback site](https://twitterdevfeedback.uservoice.com/forums/930250-twitter-api) instead.
